jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/AttachHandler.java

```diff
@@ -390,7 +390,7 @@ protected boolean terminate(boolean wakeHandler) {
 			if (LOGGING_DISABLED != loggingStatus) {
 				IPC.logMessage("AttachHandler terminate removing directory : ", TargetDirectory.getTargetDirectoryPath(getVmId())); //$NON-NLS-1$
 			}
-			TargetDirectory.deleteMyDirectory(false);
+			TargetDirectory.deleteMyDirectory(false, AttachHandler.getVmId());
```

Comment:
The reason I suggested to pass an empty string instead of passing `AttachHandler.getVmId()` is that `deleteTargetDirectory(vmId)` is only called when `if ((!directoryEmpty && !deleteMyFiles()) || (null == targetDirectoryFileObject) || !targetDirectoryFileObject.delete())`, and there is a cost to invoke `AttachHandler.getVmId()`
https://github.com/eclipse-openj9/openj9/blob/cd8db74f1d43b37e3bdbd1c72f9e120d18047634/jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/AttachHandler.java#L823-L827
Please remove it to avoid a synchronization not required.

Reply:
```suggestion
			TargetDirectory.deleteMyDirectory(false, “”);
```